cmd/admin-bucket-handlers.go

```diff
@@ -164,7 +164,12 @@ func (a adminAPIHandlers) SetRemoteTargetHandler(w http.ResponseWriter, r *http.
 		writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErrWithErr(ErrAdminConfigBadJSON, err), r.URL)
 		return
 	}
-	host, port, err := net.SplitHostPort(target.Endpoint)
+	turl := target.URL()
+	endPoint := turl.Host
+	if turl.Port() == "" {
+		endPoint = turl.Host + ":" + turl.Port()
+	}
```

Comment:
isn't this fake `turl.Port() == ""` ? the end result here is 

```
endPoint = turl.Host + ":" + turl.Port()
```

would result in 
```
"hostname:"
```

?

Reply:
this is only for validating host, port using net.SplitHostPort. Pls [see](https://github.com/golang/go/issues/25874).

Reply:
Can this be changed to
```
      if turl.Port() == "" {
		endPoint = turl.Host + ":"
	}
```